### PR TITLE
Fix scheduler type mismatch

### PR DESCRIPTION
**Pull request by pcuenca**
When doing generation manually instead of using `__call__`, and using guidance_scale as a static argument.

See #3039 for a discussion.

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion.py [hunk 1]
```diff
@@ -245,6 +245,9 @@ def _generate(
         negative_prompt_embeds = self.text_encoder(uncond_input, params=params["text_encoder"])[0]
         context = jnp.concatenate([negative_prompt_embeds, prompt_embeds])
 
+        # Ensure model output will be `float32` before going into the scheduler
+        guidance_scale = jnp.array([guidance_scale], dtype=jnp.float32)
```
Comment (pcuenca) [inline src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion.py:249]: An alternative would be to simply cast to the correct type before invoking the step function. I did it this way for symmetry with what happens here: https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion.py#L369
